### PR TITLE
Document known features on rest-api-spec tests

### PR DESCRIPTION
**Pull request by Mpdreamz**
Features dictate whether a `rest-api-spec` test runner can execute a
test. This PR documents all the know features in the java implementation
of the runner.

cc @elastic/es-clients

### CODE REVIEW

**Comment by sethmlarson** [comment]
elasticsearch-py mentions `gtelte`, is that a feature as well?

**Comment by Mpdreamz** [comment]
It used to be but was removed because all runners implemented it. 

The feature flag was there to skip tests using `lt, gt, gte, lte` operator.

File: rest-api-spec/src/main/resources/rest-api-spec/test/README.asciidoc [hunk 1]
```diff
@@ -133,6 +133,109 @@ by default, thus the related skip sections can be removed from the tests.
...
+The runner is able to set per request headers on the `do` operation
+
+=== `node_selector`
+
+Indicates the runner can parse `node_selector` under the `do` operator and use its metadata to select the node to
+perform the `do` operation on.
+
+=== `stash_in_key`
+
+Allows you use a stashed value in the any key of an object during a `match` operation
```
Comment (javanna) [inline rest-api-spec/src/main/resources/rest-api-spec/test/README.asciidoc]:
```suggestion
Allows you to use a stashed value in any key of an object during a `match` assertion
```

File: rest-api-spec/src/main/resources/rest-api-spec/test/README.asciidoc [hunk 2]
```diff
@@ -133,6 +133,109 @@ by default, thus the related skip sections can be removed from the tests.
...
+=== `embedded_stash_key`
+
+Allows a stashed key to appear anywhere in the path:
+
+....
+field1.e${placeholder}ments.element1
+....
+
+=== `stash_path_replace`
+Used only in the doc snippet tests. Allow you to do easie replacements using a special `$_path` marker.
```
Comment (javanna) [inline rest-api-spec/src/main/resources/rest-api-spec/test/README.asciidoc]:
```suggestion
Used only in the doc snippet tests. Allow you to do ease replacements using a special `$_path` marker.
```

File: rest-api-spec/src/main/resources/rest-api-spec/test/README.asciidoc [hunk 3]
```diff
@@ -133,6 +133,109 @@ by default, thus the related skip sections can be removed from the tests.
...
+
+Allows a stashed value to be referenced in path lookups as a single token. E.g:
+
+....
+path.$stash.value
+....
+
+=== `embedded_stash_key`
+
+Allows a stashed key to appear anywhere in the path:
```
Comment (javanna) [inline rest-api-spec/src/main/resources/rest-api-spec/test/README.asciidoc]:
```suggestion
Allows a stashed key to appear anywhere in the path (note the placeholder needs to be within curly brackets too in this case):
```

**Review by javanna — APPROVED** [review]
I added a couple of suggestions, LGTM otherwise

**Comment by Mpdreamz** [comment]
Thanks for the notes @javanna 👍

**Pull request merged**